Services/Devices.cs

```diff
@@ -403,6 +519,67 @@ private int AnalyzeBatchErrors(BulkRegistryOperationResult result)
             return errorsByType.Count;
         }
 
+        private async Task<CloudBlockBlob> WriteDevicesToBlobAsync(List<string> serializedDevices)
+        {
+            var sb = new StringBuilder();
+            serializedDevices.ForEach(serializedDevice => sb.AppendLine(serializedDevice));
+
+            // Write to blob
+            var blob = await this.CreateImportExportBlobAsync();
+            using (var stream = await blob.OpenWriteAsync())
+            {
+                byte[] bytes = Encoding.UTF8.GetBytes(sb.ToString());
+                for (var i = 0; i < bytes.Length; i += 500)
+                {
+                    int length = Math.Min(bytes.Length - i, 500);
+                    await stream.WriteAsync(bytes, i, length);
+                }
+            }
+
+            return blob;
+        }
+
+        private async Task<CloudBlockBlob> CreateImportExportBlobAsync()
+        {
+            // Container for the files managed by Azure IoT SDK.
+            // Note: use a new container to speed up the operation and avoid old files left over
+            string containerName = ("iothub-" + DateTimeOffset.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss-") + Guid.NewGuid().ToString("N")).ToLowerInvariant();
+            string blobName = "devices.txt";
+
+            this.log.Info("Creating import blob", () => new { containerName, blobName });
+
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(this.config.IoTHubImportStorageAccount);
+            CloudBlobClient client = storageAccount.CreateCloudBlobClient();
+            CloudBlobContainer container = client.GetContainerReference(containerName);
+            CloudBlockBlob blob = container.GetBlockBlobReference(blobName);
+
+            await container.CreateIfNotExistsAsync();
+            await blob.DeleteIfExistsAsync();
```

Comment:
Why are we deleting the blob right after creating it?

Reply:
container != blob.  The container is created, and it should be empty.

Reply:
Why do we need to delete the blob from a new and empty container?